src/oc/change/async/persistence.clj

```diff
@@ -93,17 +117,19 @@
   ;; READS
 
   ([message :guard :status]
-  ;; Lookup when a specified user saw specified containers and when the specified containers saw changes
-  ;; Send the merger of the 2 (by container-id) to the sender's channel as a container/status message
+  ;; Lookup when a specified user saw specified containers, when the specified containers saw changes,
+  ;; and when the specified user had reads in specified containers.
+  ;; Send the processed result to the sender's channel as a container/status message
   (let [user-id (:user-id message)
         container-ids (:container-ids message)
         client-id (:client-id message)
         just-seen (:just-seen message)] ; an entry just written to the DB that might not be readable yet
     (timbre/info "Status request for:" container-ids "by:" user-id "/" client-id)
     (let [seens (filter #((set container-ids) (:container-id %)) (seen/retrieve user-id))
           all-seens (if just-seen (conj seens just-seen) seens) ; avoid a race condition in the DB
+          all-reads (filter #((set container-ids) (:container-id %)) (read/retrieve-by-user user-id))
```

Comment:
@belucid I'm curious, now that there is a GSI of `[:user-id :container-id]` could this computation be done in one fell swoop in a single query? Perhaps if `retrieve-by-user` had a 2-arity variant where the second arg was a container-id set, it could include that set in the query. There is most likely some technical DynamoDB reason that this can't be done that I'm unaware of...

Reply:
@cjsauer I think that's a good catch. Looking into it.

Reply:
Ah, after looking into it, it turns out we can have a 2nd arity fn that does this (and I wrote it, so left it in), but we can't actually use it here. Reason being is we have a whole set of container IDs we and Dynamo doesn't have an `IN` style query: `comparison-operators e/o #{:eq :le :lt :ge :gt :begins-with :between}`

So to use the new 2/ariy fn we'd have to do multiple queries, one for each container we have. More efficient to do the 1 query and filter by the containers as we were doing.